pkg/kustomize/apply.go

```diff
@@ -21,6 +22,7 @@ import (
 const (
 	retryInterval = 10 * time.Second
 	retryTimeout  = 1 * time.Minute
+	retryMaxCount = 3
```

Comment:
Why 3? Should we retry at all?

Reply:
Can the apply operation take time? We have timeouts and retries to mitigate that, I think.

Reply:
I'm not sure why any given API call would take very long. The resource might not be updated immediately (a Deployment might not produce a Pod right away, for example), but the API call itself should return quickly.

How complex do we want this to be? Should we have some sort of backoff algorithm for the delay between retries? Should it retry indefinitely (with that extra delay), like a normal controller, in case some agent updates the files?

Reply:
We can use different versions of Poll function to implement poll techniques according to our choice.
I would leave this with a few retries and 10s delay in between and see if we get more complains from the end-users.

Reply:
@ggiguash  My understanding of the `indefinite retry` is that microshift logs error and still proceeds when kustomize fails, but it comes back after the syncInterval and run kustomize again with the hope that user agent may fix the bad yamls.

Reply:
I wonder if this scenario is useful on edge devices. If an update is installed affecting the manifests, the reboot needs to take place anyway, right?

Reply:
@dhellmann / @zshi-redhat 
To make progress with this PR, I need your recommendation on the number of retries. We can set it to 1, 3, or -1 for indefinite.

Reply:
@ggiguash  It may not be just about changing the retry value.
I wonder if we could implement an indefinite retry on the kustomize service controller, rather than inside the kustomize apply function. It means that microshift will proceed to start the rest of service controllers (kubelet, load balancer service controllers) after the first call of kustomize fails, but be able to rerun the kustomize service controller based on the last kustomize return value.

Reply:
I'm not sure if this approach is necessary for edge devices where the updates actually require rebooting the host. If we're rebooting, the indefinite retry seems to be redundant, doesn't it?